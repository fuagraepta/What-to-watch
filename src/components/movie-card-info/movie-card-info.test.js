import React from 'react';
import renderer from 'react-test-renderer';

import MovieCardInfo from './movie-card-info.jsx';

const mockComponent = <div>Hello World</div>;

describe(`The component is rendered correctly`, () => {
  it(`Movie-card-info correctly renders with transferred mock data`, () => {
    const movieCardInfo = renderer
      .create(
          <MovieCardInfo
            poster = {`img/the-grand-budapest-hotel-poster.jpg`}
            title = {`Johnny English`}
            isMoviePage = {false}
          >
            {mockComponent}
          </MovieCardInfo>
      )
      .toJSON();

    expect(movieCardInfo).toMatchSnapshot();
  });

  it(`Movie-card-info correctly renders with transferred isMoviePage prop`, () => {
    const movieCardInfo = renderer
      .create(
          <MovieCardInfo
            poster = {`img/the-grand-budapest-hotel-poster.jpg`}
            title = {`Johnny English`}
            isMoviePage = {true}
          >
            {mockComponent}
          </MovieCardInfo>
      )
      .toJSON();

    expect(movieCardInfo).toMatchSnapshot();
  });
});
