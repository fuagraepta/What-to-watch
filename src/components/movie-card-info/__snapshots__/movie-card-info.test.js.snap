// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The component is rendered correctly Movie-card-info correctly renders with transferred isMoviePage prop 1`] = `
<div
  className="movie-card__info"
>
  <div
    className="movie-card__poster movie-card__poster--big"
  >
    <img
      alt="Johnny English"
      height="327"
      src="img/the-grand-budapest-hotel-poster.jpg"
      width="218"
    />
  </div>
  <div>
    Hello World
  </div>
</div>
`;

exports[`The component is rendered correctly Movie-card-info correctly renders with transferred mock data 1`] = `
<div
  className="movie-card__info"
>
  <div
    className="movie-card__poster "
  >
    <img
      alt="Johnny English"
      height="327"
      src="img/the-grand-budapest-hotel-poster.jpg"
      width="218"
    />
  </div>
  <div>
    Hello World
  </div>
</div>
`;
